framework/src/wifi_manager/wifi_manager_state.c

```diff
@@ -136,10 +136,13 @@ static char *wifimgr_state_str[] = {
 #define WIFIMGR_GET_STATE g_manager_info.state
 #define WIFIMGR_GET_PREVSTATE g_manager_info.prev_state
 #define WIFIMGR_STORE_PREV_STATE (g_manager_info.prev_state = g_manager_info.state)
-#define WIFIMGR_RESTORE_STATE								\
-	do {													\
-		g_manager_info.state = g_manager_info.prev_state;	\
-		g_manager_info.prev_state = WIFIMGR_NONE;			\
+#define WIFIMGR_RESTORE_STATE                             \
+	do {                                                  \
+		g_manager_info.state = g_manager_info.prev_state; \
+		g_manager_info.prev_state = WIFIMGR_NONE;         \
+		wifimgr_info_msg_s twmsg;                         \
+		twmsg.state = g_manager_info.state;               \
+		wifimgr_set_info(WIFIMGR_STATE, &twmsg);         \
```

Comment:
In this case,
`g_manager_info.state` and `g_manager_info.prev_state` are set as same, the original `g_manager_info.prev_state`.

Let's assume, the original `g_manager_info.prev_state` is `A`.
```
line 141 : g_manager_info.state = A
line 142 : g_manager_info.prev_state = WIFIMGR_NONE
line 144 : twmsg.state  = A (Because of line 141)
line 145 : In wifimgr_set_info(), 
                 g_manager_info.prev_state = g_manager_info.state (== A)
                 g_manager_info.state = info->state(passed param) (== A)
```
Then, `g_manager_info.state` and `g_manager_info.prev_state` are all `A`, which is the original `g_manager_info.prev_state` .

So I think, this macro is as same as just setting `g_manager_info.state = g_manager_info.prev_state`.
How do you think?

Reply:
g_manger_info in wifi_manager_state.c and g_manager_info in wifi_manager_info.c are different. So prev_state is not overwritten.

Reply:
@pillip8282 I understand two static variables have different purposes: one for state and the other for information.
However, it should've been different for better understanding.